crates/ruma-identity-service-api/src/association/unbind_3pid/v2.rs

```diff
@@ -0,0 +1,94 @@
+//! [POST /_matrix/identity/v2/3pid/unbind](https://matrix.org/docs/spec/identity_service/r0.3.0#post-matrix-identity-v2-3pid-unbind)
+
+use ruma_api::ruma_api;
+use ruma_common::thirdparty::Medium;
+use ruma_identifiers::{user_id::UserId, ClientSecretBox, SessionIdBox};
+use serde::{Deserialize, Serialize};
+
+ruma_api! {
+    metadata: {
+        description: "Remove an association between a session and a Matrix user ID.",
+        method: POST,
+        name: "unbind_3pid",
+        path: "/_matrix/identity/v2/3pid/unbind",
+        rate_limited: false,
+        authentication: AccessToken,
+    }
+
+    request: {
+        /// The proof that the client owns the 3PID.
+        ///
+        /// If this is not provided, the request must be signed by the homeserver which controls
+        /// the `user_id`.
```

Comment:
```suggestion
        /// the `mxid`.
```
`user_id` is mentioned in the docs, but it's not the name of any parameter on this endpoint. I _think_ it refers to the `mxid` parameter, but I can't really think this morning. Does that make sense to you? (A `git blame` on this part of the spec might show if there was mass parameter rename or something but this was missed in the docs.)

Either way, we should modify this doc to refer to the parameter name.

Reply:
Changed to `mxid`, I also don't see what else it could refer to and it does make sene. I did have a look at the history of the IS spec, and it seems that the inconsistency was introduced when v1 of this endpoint was specced, and copied word-for-word in the v2 endpoint.

Reply:
Found the answer in the [MSC which introduced the endpoint](https://github.com/matrix-org/matrix-doc/blob/master/proposals/1915-unbind-identity-server-param.md#identity-server-3pid-unbind-api):

> Add `POST /_matrix/identity/api/v1/3pid/unbind` with `mxid` and `threepid` fields.
> The `mxid` is the user's `user_id` and `threepid` is a dict with the usual
> `medium` and `address` fields.
> 
> If the server returns a 400, 404 or 501 HTTP error code then the homeserver
> should assume that the identity server doesn't support the `/3pid/unbind` API, unless
> it returns a specific matrix error response (i.e. the body is a JSON object with
> `error` and `errcode` fields).
> 
> The identity server should authenticate the request in one of two ways:
> 
> 1. The request is signed by the homeserver which controls the `user_id`.
> 2. The request includes the `sid` and `client_secret` params (as per `/bind`),
>    which proves ownership of the given 3PID.
> 

Or to be more concise:
> The `mxid` is the user's `user_id`